x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/Transform.java

```diff
@@ -131,16 +132,17 @@ public class Transform extends Plugin implements SystemIndexPlugin, PersistentTa
     private final Settings settings;
     private final SetOnce<TransformServices> transformServices = new SetOnce<>();
 
-    public static final int DEFAULT_FAILURE_RETRIES = 10;
     public static final Integer DEFAULT_INITIAL_MAX_PAGE_SEARCH_SIZE = Integer.valueOf(500);
     public static final TimeValue DEFAULT_TRANSFORM_FREQUENCY = TimeValue.timeValueMillis(60000);
 
-    // How many times the transform task can retry on an non-critical failure
+    public static final int DEFAULT_FAILURE_RETRIES = 10;
+    // How many times the transform task can retry on a non-critical failure.
+    // This cluster-level setting can be overriden by transform-level setting specified in individual transform's SettingsConfig.
     public static final Setting<Integer> NUM_FAILURE_RETRIES_SETTING = Setting.intSetting(
```

Comment:
I think we should discourage the use of this cluster setting and mark it deprecated.

Reply:
Done (I used `Property.DeprecatedWarning` for that).

Reply:
As suggested by Dave R., deprecating a cluster setting is a little more involved.
Because of that I revert this change and leave it as a follup.